From 1bcc5dfe60b5dd9c9285c68183f85d934ed3ef66 Mon Sep 17 00:00:00 2001
From: Octavian Purdila <octavian.purdila@nxp.com>
Date: Fri, 3 Feb 2017 15:41:08 +0200
Subject: [PATCH 1392/5242] MLK-13859: video: mxc: mipi dsi: don't print
 errors on EPROBE_DEFER

commit  792273dbafc2716ad8e8bde9fa3b263994641e80 from
https://source.codeaurora.org/external/imx/linux-imx.git

The device may fail to reset with EPROBE_DEFER because of the
initialization order between the mipi and the mipi gpio reset
drivers. This is not an error, since the probe will be later retried
and eventually will succeed.

Signed-off-by: Octavian Purdila <octavian.purdila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi.c b/drivers/video/fbdev/mxc/mipi_dsi.c
index 3e78a53..d17d19a 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi.c
@@ -872,7 +872,8 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 
 	ret = device_reset(&pdev->dev);
 	if (ret) {
-		dev_err(&pdev->dev, "failed to reset: %d\n", ret);
+		if (ret != -EPROBE_DEFER)
+			dev_err(&pdev->dev, "failed to reset: %d\n", ret);
 		goto dev_reset_fail;
 	}
 
-- 
1.7.9.5

